source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -410,6 +413,75 @@ json TransformerInstance::base64url_decode_callback(const inja::Arguments &args)
   return Base64Url::decode(input);
 }
 
+json TransformerInstance::word_count_callback(const inja::Arguments &args) {
+  return std::to_string(json_word_count(args.at(0)));
+}
+
+int TransformerInstance::json_word_count(const nlohmann::json* input)  {
+  if (input->is_string()) {
+    const std::string &input_string = input->get_ref<const std::string &>();
+    return word_count(input_string);
+  } else if (input->is_array()) {
+    int total_word_count = 0;
+    const auto &input_array = input->get_ref<const std::vector<json> &>();
+    for (auto & element : input_array) {
+      total_word_count += json_word_count(&element);
+    }
+    return total_word_count;
+  } else if (input->is_object()) {
+    int total_word_count = 0;
+    const auto element_obj = input->get_ref<const json::object_t &>();
+    for (auto & [key, value] : element_obj) {
+      total_word_count += word_count(key);
+      total_word_count += json_word_count(&value);
+    }
+    return total_word_count;
+  } else if (input->is_number() || input->is_boolean()) {
+    // Booleans and numbers are constant
+    return 1;
+  }
```

Comment:
do we want to add a debug assert / log if nothing matches?

Reply:
I don't think so, if you pass in some empty struct the result may be zero, but we already have logs to show the inputs and things so that should be enough to deduce.